otherlibs/stdlib_alpha/effect.ml

```diff
@@ -577,10 +579,45 @@ let run_with (type a e es) (local_ hs : es Handler.List.t)
   let res = run_stack (module H) f handlers in
   handle mapping res
 
+module type Aux = sig
+  val fiber : 'a 'b 'e . ('e Handler.t @ local -> 'a -> 'b)
+    -> ('a, 'b, 'e, unit) continuation
+
+  val fiber_with : 'a 'b 'e 'es . 'es Handler.List.Length.t @ local
+    -> (('e * 'es) Handler.List.t @ local -> 'a -> 'b)
+    -> ('a, 'b, 'e, 'es) continuation
+
+  val run : 'a 'e . ('e Handler.t @ local -> 'a)
+    -> ('a, 'e, unit) res
+
+  val run_with : 'a 'e 'es . 'es Handler.List.t @ local
+    -> (('e * 'es) Handler.List.t @ local -> 'a)
+    -> ('a, 'e, 'es) res
+end
+
+module type PortableAux = sig include Aux @@ portable end
+let magic_aux (aux : (module Aux)) : (module PortableAux) = Obj.magic aux
+
+(* CR dkalinichenko: [fiber], [fiber_wth], [run] and [run_with] are not
+   [portable] because they use first-class modules, even though it's safe
+   to declare them [portable] becuase the first-class module is created locally.
```

Comment:
I think they actually can't be portable due to the use of the `Handler.Create` functor, since functors cannot yet be portable. That functor can be replaced with a function that takes a first-class module and returns another first-class mode. Such a function can be portable, so I think that might be enough to remove this magic.

Reply:
Done. Unfortunately, creating the first-class module directly in the function leads to inefficient compilation, making one of the tests fail running out of memory. But with the functor private to the implementation, we can limit the magic to one function.

Reply:
The allocation semantics of the functor and the function working on first-class modules should actually be equivalent, so I don't think that should actually happen. Functors do get more aggressive inlining, maybe annotating the function `[@inline]` would fix it. Either way we shouldn't be able to cause anything to run out of memory -- any such allocation would be extremely short-lived. I think it would be worth working out what was going on there.

Reply:
The exact error is `Fatal error: allocation failure during minor GC`, and it happens only in bytecode in the `evenodd.ml` test, which creates 100000 handlers in succession.

Reply:
Seems like it doesn't happen in the test runner, but I can consistently reproduce it on my machine.

Reply:
Looks like the value handler pointer in the top ~25000 handlers is for some reason not updated after a major GC in bytecode, which leads to a crash when trying to interpret them as pointers to live blocks. I'll try to reproduce this using upstream effects.

Reply:
Max pointed me to https://github.com/ocaml-flambda/flambda-backend/pull/3231, which allowed me to resolve the issue. I'll create a separate PR for it.